### ISSUE TITLE
Quote hostnames in default ingress YAML

### ISSUE THREAD

**Issue by frasern** [kubernetes/helm#3936]
Wildcard hostnames such as ***.example.com** break the default ingress created by `helm create` because YAML treats elements starting with an asterisk specially.

The values therefore need to be quoted in order to be recognised as strings.

**Comment by jascott1** [comment]
Hi @frasern 

i have confirmed the behavior but am unable to fix it in the template. Quoting in the values seems to resolve the issue. Do you have a template based fix?

**Comment by jascott1** [comment]
I tried that before posting my last comment and it didn't work. Did you have success?

**Comment by frasern** [comment]
Hi @jascott1 -- yes, it seemed to work for me (using helm v2.8.2).

Here's what I did:

1. Create a test chart:
```bash
$ helm create test
```

2. Edit `test/values.yaml`, setting the ingress to be enabled and to have wildcard names:
```yaml
ingress:
  enabled: true
  annotations: {}
    # kubernetes.io/ingress.class: nginx
    # kubernetes.io/tls-acme: "true"
  path: /
  hosts:
    - "*.chart-example.local"
  tls:
    - secretName: chart-example-tls
      hosts:
        - "*.chart-example.local"
```

3. Run `helm install --dry-run --debug test` and note the ingress definition has non-quoted hostnames:

```yaml
# Source: test/templates/ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: kissed-pug-test
  labels:
    app: test
    chart: test-0.1.0
    release: kissed-pug
    heritage: Tiller
spec:
  tls:
    - hosts:
        - *.chart-example.local
      secretName: chart-example-tls
  rules:
    - host: *.chart-example.local
      http:
        paths:
          - path: /
            backend:
              serviceName: kissed-pug-test
              servicePort: http
```

4. Edit `test/templates/ingress.yaml` to change the two `{{ . }}` entries to `{{ . | quote }}`.

5. Re-run `helm install --dry-run --debug test` and note the ingress definition now has correctly quoted hostnames:

```yaml
# Source: test/templates/ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: happy-mink-test
  labels:
    app: test
    chart: test-0.1.0
    release: happy-mink
    heritage: Tiller
spec:
  tls:
    - hosts:
        - "*.chart-example.local"
      secretName: chart-example-tls
  rules:
    - host: "*.chart-example.local"
      http:
        paths:
          - path: /
            backend:
              serviceName: happy-mink-test
              servicePort: http
```

**Comment by jstoja** [comment]
@frasern Do you want to create a PR for that or someone can implement it with your findings?

**Comment by bacongobbler** [comment]
closed via #3956
